test/integration/job/job_test.go

```diff
@@ -1749,6 +1757,32 @@ func TestJobPodReplacementPolicy(t *testing.T) {
 			},
 			wantActive: int(podCount),
 		},
+		"feature flag true, delete pods, verify active and failed status and recreate once failed": {
+			podReplacementPolicyEnabled: true,
+			failPods:                    true,
+			jobSpec: &batchv1.JobSpec{
+				Parallelism:          pointer.Int32Ptr(podCount),
+				Completions:          pointer.Int32Ptr(podCount),
+				CompletionMode:       &indexedCompletion,
+				PodReplacementPolicy: podReplacementPolicy(batchv1.Failed),
+			},
+			wantActive:      int(podCount),
+			wantFailed:      int(podCount),
+			wantTerminating: pointer.Int32(0),
+		},
+		"feature flag true with NonIndexedJob, delete pods, verify active and failed status and recreate once failed": {
```

Comment:
```suggestion
		"feature flag true with NonIndexedJob, recreate failed pods, and verify active and failed counters": {
```
nit, as in the other case.